DON'T MERGE: Fix Docker configuration to work correctly with openidc-baseimage

coreone opened the pull request:
The `run.sh` script that comes with openidc-baseimage is pretty important, so this PR is to keep the `run.sh` in place while still allowing customizations via the `override.sh` script.

coreone committed:
* Rename run-apache.sh to override.sh for use with new openidc-baseimage image
* Add new files to Dockerfile
* Change run script to selectively choosing the correct site config instead
  of generating the site file on the fly
* Add locations.conf file as a common area to hold the Locations settings

coreone committed:
* Rename run-apache.sh to override.sh for use with new openidc-baseimage image
* Add new files to Dockerfile
* Change run script to selectively choosing the correct site config instead
  of generating the site file on the fly
* Add locations.conf file as a common area to hold the Locations settings

coreone committed:
Make sure that if HTTPS_ONLY changes, the config will revert back to site.conf

coreone committed:
Make sure that if HTTPS_ONLY changes, the config will revert back to site.conf

dmohs committed:
Add "link NIH account" link to profile page

coreone committed:
Remove SSLProxyEngine in https-only config

coreone committed:
Remove SSLProxyEngine in https-only config

coreone committed:
Put openidc-baseimage back to 1.8.6

coreone committed:
Put openidc-baseimage back to 1.8.6

coreone committed:
Add higher security options to site.conf and site-https-only.conf

coreone committed:
Add higher security options to site.conf and site-https-only.conf

coreone committed:
Add X_FRAME_OPTIONS option to site-https-only.conf and site.conf

coreone committed:
Add X_FRAME_OPTIONS option to site-https-only.conf and site.conf

dmohs commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM broadinstitute/openidc-baseimage:1.8.5
+FROM broadinstitute/openidc-baseimage:1.8.6
```
Since we're not using any of the authentication bits, maybe `FROM httpd:2.4` is more appropriate?

dmohs committed:
Add configurable shibboleth URL

dmohs committed:
Fix infinite loop on error response

dmohs committed:
Dev tools tweaks and one addition

dmohs committed:
Merge pull request #340 from broadinstitute/dmohs-nih-link-DSDEEPB-1925

Link NIH account (DSDEEPB-1925)

dmohs committed:
Update to React 14
- fixes all usages of getDOMNode
- each page touched tested manually

dmohs committed:
Exclude config.json from clean script

davidangb committed:
Merge pull request #341 from broadinstitute/dmohs-react-update

Update to React 14

davidangb committed:
fix regression with react upgrade

dmohs committed:
Merge pull request #342 from broadinstitute/da_reactUpgradeFix

fix regression with react upgrade

dmohs committed:
Bump dmohs/react version

dmohs committed:
Call orchestration endpoint to link NIH account

davidangb committed:
Merge pull request #343 from broadinstitute/dmohs-nih-link

Call orchestration endpoint to link NIH account

davidangb committed:
use value of billing project, not option index, when creating workspace.

davidangb committed:
use value of billing project, not option index, when cloning workspace.

davidangb committed:
add CSS overflow to GCSFilePreviewLink

rushtong committed:
Merge pull request #344 from broadinstitute/da_DSDEEPB-2284_projectDropdown

DSDEEPB-2284: create/clong workspace billing project dropdown

rushtong committed:
Merge pull request #345 from broadinstitute/da_DSDEEPB-2439_overflowGCSFilePreviewLink

DSDEEPB-2439: add CSS overflow to GCSFilePreviewLink

rushtong commented:
FYI, the build for this image is hanging for me and never gets past setting up certificates:
```
...
Creating config file /etc/php5/cli/php.ini with new version
php5_invoke pdo: already enabled for cli SAPI
php5_invoke opcache: already enabled for cli SAPI
php5_invoke json: already enabled for cli SAPI
Setting up libnss3-nssdb (2:3.19.2.1-0ubuntu0.14.04.1) ...
Setting up libnss3:amd64 (2:3.19.2.1-0ubuntu0.14.04.1) ...
Setting up openjdk-8-jre-headless:amd64 (8u72-b05-5~trusty1) ...
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java to provide /usr/bin/java (java) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/jjs to provide /usr/bin/jjs (jjs) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/keytool to provide /usr/bin/keytool (keytool) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/pack200 to provide /usr/bin/pack200 (pack200) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/rmid to provide /usr/bin/rmid (rmid) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/rmiregistry to provide /usr/bin/rmiregistry (rmiregistry) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/unpack200 to provide /usr/bin/unpack200 (unpack200) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/orbd to provide /usr/bin/orbd (orbd) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/servertool to provide /usr/bin/servertool (servertool) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/tnameserv to provide /usr/bin/tnameserv (tnameserv) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/jexec to provide /usr/bin/jexec (jexec) in auto mode
Setting up libatk-wrapper-java (0.30.4-4) ...
Setting up ca-certificates-java (20130815ubuntu1) ...
```

davidangb commented:
@coreone is this ready to go? Before the holiday break I saw a note to hold off on this ...

coreone commented:
Yes, this should be good now.  We were working out some things with openidc-baseimage, but they have been sorted, so it should be good to go.

rushtong committed:
Add mac file patterns to .gitignore

Commit:
Improve 401 check to use new /me endpoint

MattPutnam committed:
Merge pull request #346 from broadinstitute/gr-add-osx-to-ignore

Add mac file patterns to .gitignore

davidangb committed:
use dropdown for root entity type

Commit:
Implement full dialog for method/config redact, with error viewer for 401s

davidangb committed:
Merge pull request #347 from broadinstitute/mp/gr-update401check

Improve 401 check to use new /me endpoint

helgridly committed:
removed box-shadow from table paginator because it's not used anywhere else except very subtly on text boxes. so i removed it there too.

davidangb committed:
Merge pull request #349 from broadinstitute/mp-DSDEEPB-2047

DSDEEPB-2047: Implement full dialog for method/config redact, with error viewer for 401s

dmohs committed:
NIH account link timeout warning
Also some minor UI fixes

davidangb committed:
PR feedback: indentation; use a vector instead of a list; use passed argument instead of hardcoded value

MattPutnam committed:
Merge pull request #348 from broadinstitute/da_DSDEEPB-1929_rootEntityTypeDropdown

DSDEEPB-1929: use dropdown for root entity type

Commit:
Overhaul method/config ACL editor

* Use validatable input fields, and error displaying utilities
* Put UI in the OKCancelForm
* Significant cleanup throughout

Commit:
Added Cromwell's Aborting status

dmohs committed:
Address review comments

MattPutnam committed:
Merge pull request #350 from helgridly/no_shadow

killed box-shadow

davidangb committed:
Merge pull request #352 from broadinstitute/dmohs-warning-DSDEEPB-2280

NIH account link timeout warning

davidangb committed:
use version number for Thurloe profile instead of true/false

Commit:
Merge pull request #353 from broadinstitute/aborting_status

Added Cromwell's Aborting status

Commit:
DSDEEPB-2266: Use /inputsOutputs endpoint and fix method config edit

* Using proper endpoint to get all inputs and outputs, not just ones that have values
* Don't save empty strings, instead remove the entry from the payload
* Display input/output type and if it's optional
* General cleanup of the editor code

rushtong committed:
Show inputs with workflow submission errors

rushtong committed:
Merge pull request #355 from broadinstitute/da_DSDEEPB-2248_profileVersionTag

DSDEEPB-2248: use version number for Thurloe profile instead of true/false

dmohs committed:
Fix deletion bug

dmohs committed:
Merge pull request #354 from broadinstitute/mp-DSDEEPB-2048

DSDEEPB-2048: Overhaul method/config ACL editor

rushtong committed:
Add mock data for input resolutions
Refactor convenience method to get keys from input resolutions

davidangb committed:
Merge pull request #358 from broadinstitute/dmohs-delete-bug-DSDEEPB-2351

Fix deletion bug

davidangb committed:
Merge pull request #356 from broadinstitute/gr-DSDEEPB-2011

Show inputs with workflow submission errors

Commit:
DSDEEPB-2045: Added help text and validation predicates to workspace name

dmohs committed:
Merge pull request #357 from broadinstitute/mp-DSDEEPB-2266

DSDEEPB-2266: Use /inputsOutputs endpoint and fix method config edit

davidangb committed:
Merge pull request #359 from broadinstitute/mp-DSDEEPB-2045

DSDEEPB-2045: Added help text and validation predicates to workspace name

Commit:
DSDEEPB-2393: Fix ordering bug

davidangb committed:
Merge pull request #360 from broadinstitute/mp-DSDEEPB-2393

DSDEEPB-2393: Fix ordering bug

dmohs committed:
Fix Workspace ACL editing

Commit:
DSDEEPB-2381: fix status icons

* Use "executionStatus" (cromwell status) instead of "backendStatus" (JES status) for call-level detail
* Separate workflow vs. call statuses, centralize lists of values, and warn if match not found

davidangb committed:
Merge pull request #361 from broadinstitute/dmohs-ws-acl-DSDEEPB-2461

Fix Workspace ACL editing

davidangb committed:
Merge pull request #364 from broadinstitute/mp-DSDEEPB-2381

DSDEEPB-2381: fix status icons

davidangb committed:
dbGaP, not dbGAP

Commit:
Show error if method can't be found for a method config

rushtong committed:
UI work for adding new fields to the profile page

dmohs committed:
Merge pull request #365 from broadinstitute/mp-methodIO-fix

Show error if method can't be found for a method config

Commit:
Fix react sync issue with table sorting

dmohs committed:
Merge pull request #366 from broadinstitute/mp-table-sort

Fix react sync issue with table sorting

rushtong committed:
Change the display of submission status counts

Commit:
add help forum link

davidangb committed:
Merge pull request #368 from broadinstitute/mp-DSDEEPB-2533

DSDEEPB-2533: Point "Support" footer link to the forum

dmohs committed:
Security fixes per DSDEEPB-2534

davidangb committed:
better error message for inactive users

davidangb committed:
add config.json for prod

MattPutnam committed:
Merge pull request #371 from broadinstitute/da_DSDEEPB-2462_unactivatedErrorMessage

DSDEEPB-2462: better error message for inactive users

davidangb committed:
Merge pull request #372 from broadinstitute/da_shibProd

add config.json for prod

davidangb committed:
Merge pull request #370 from broadinstitute/dmohs-apache-security-DSDEEPB-2534

Security fixes per DSDEEPB-2534

rushtong committed:
Update NIH link expiration text

davidangb committed:
layout tweaks to nih warning banner

rushtong committed:
Merge pull request #375 from broadinstitute/da-DSDEEPB-2579

layout tweaks to nih warning banner

davidangb committed:
Merge pull request #374 from broadinstitute/gr-DSDEEPB-2579

Update NIH link expiration text

davidangb committed:
update system use notification message verbiage

dmohs committed:
Refactor profile UI

davidangb committed:
further updates to NIH link warning banner verbiage

rushtong committed:
Add text-field validation error handling

rushtong committed:
Merge pull request #376 from broadinstitute/da_DSDEEPB-2064_systemUseNotification

DSDEEPB-2064: update system use notification message verbiage

rushtong committed:
Merge pull request #378 from broadinstitute/da_DSDEEPB-2579_linkWarningBannerTake3

DSDEEPB-2579: further updates to NIH link warning banner verbiage

MattPutnam committed:
Merge pull request #379 from broadinstitute/gr-ui-DSDEEPB-2447

New User Profile Fields

MattPutnam committed:
Merge pull request #373 from broadinstitute/gr-DSDEEPB-2476

Change the display of submission status counts

dmohs committed:
Address review comments

davidangb committed:
Merge pull request #377 from broadinstitute/dmohs-profile-DSDEEPB-2404

Refactor profile UI

abaumann committed:
NOTE: this does NOT need to be merged prior to 1/20, but can be if we
cut another RC.

We will only have two instances of shibboleth that we ever refer to:
production and "dev".  Dev will be used by ANY environment that is
not prod to test againsnt a mocked out shibboleth server.

davidangb committed:
use supported url for GCS downloads, a la https://cloud.google.com/storage/docs/authentication#cookieauth

Commit:
DSDEEPB-2269: Reload page with new name on rename

Commit:
Render date as 'Pending...' if value not present

Commit:
DSDEEPB-2373: Don't show 'Failed to Start' div if list is empty

dmohs committed:
Merge pull request #381 from broadinstitute/da_GCSDownloadUrl

use supported url for GCS downloads

dmohs committed:
Merge pull request #382 from broadinstitute/mp-DSDEEPB-2249

DSDEEPB-2249: Render date as 'Pending...' if value not present

dmohs committed:
Merge pull request #383 from broadinstitute/mp-DSDEEPB-2373

DSDEEPB-2373: Don't show 'Failed to Start' div if list is empty

dmohs committed:
Merge pull request #384 from broadinstitute/mp-DSDEEPB-2269

DSDEEPB-2269: Reload page with new name on rename

Commit:
Fix layout issues in workspaces list table

Commit:
DSDEEPB-2240: Make proper links for nav-changing clickables

davidangb committed:
Merge pull request #380 from broadinstitute/ab_internal_envs_use_shibb_dev

No rush to merge: change shibboleth server url to .dev for alpha, staging

davidangb commented:
Finally getting back to looking at this. I was able to build the image, but after starting it (via uiorch-compose.yaml in firecloud-environment), I do not see FireCloud. Apache says it is started, but it's not serving any content. Did this actually work for you? I haven't tried to debug it. @coreone

davidangb committed:
Merge pull request #386 from broadinstitute/mp-DSDEEPB-2204

DSDEEPB-2204: Fix layout issues in workspaces list table

davidangb committed:
Merge pull request #387 from broadinstitute/mp-DSDEEPB-2240

DSDEEPB-2240: Make proper links for nav-changing clickables

dmohs commented:
I suggest we close this PR and instead create a PR to move to a standard Apache or Nginx proxy.

coreone commented:
@dmohs that makes sense.  We have a straight "enforce HTTPS" apache container currently that we use on our Jenkins hosts and other things.  Maybe we can do some updates on that to get this working for you in a more streamlined fashion.

dmohs commented:
@coreone We might be able to use that. Right now, we use HTTP for local development because Figwheel connects over an HTTP websocket to port 3449 and Chrome will not allow that if the page is HTTPS. Updating the Figwheel code to support HTTPS would be a long-term goal.

However, we might be able to proxy the websocket port so we can use HTTPS. It's something I've been wanting to look into.

coreone committed:
Make sure containers restart if Docker restarts or if they fall over

coreone committed:
Make sure containers restart if Docker restarts or if they fall over

coreone committed:
Merge branch 'tex_baseimage_fix' of github.com:broadinstitute/firecloud-ui into tex_baseimage_fix

dmohs commented:
I'm pairing with @coreone to redo this using the updated configuration architecture. Closing.
